Zend/zend_compile.c

```diff
@@ -7365,6 +7374,52 @@ void zend_compile_class_decl(znode *result, zend_ast *ast, zend_bool toplevel) /
 			zend_resolve_const_class_name_reference(extends_ast, "class name");
 	}
 
+	if (enum_scalar_type_ast != NULL) {
```

Comment:
The logic here is scattered around the file, can/should it be moved to zend_enum.c?